packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx

```diff
@@ -19,91 +19,114 @@ import { palleteColors } from './ColorPickerUtils/TagColors';
 
 export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps): JSX.Element => {
   const [showPicker, setShowPicker] = useState<boolean>(false);
+  const [newColor, setNewColor] = useState<string>(color);
   const [showChromePicker, setShowChromePicker] = useState<boolean>(false);
+  const [hexCodeError, setHexCodeError] = useState<string>(''); // State variable for hex code error
+
   const intl = useIntl();
 
-  const handleClick = useCallback(() => {
+  /**
+   * This method uses a regular expression (`hexRegex`) to validate a hex color code.
+   * The regex checks if the hex code starts with a "#" symbol, followed by either a
+   * 6-digit or 3-digit combination of characters from A-F, a-f, and 0-9.
+   * The `test` method is then used to validate the `hexCode` against the regex pattern.
+   * @param hexCode
+   * @returns
+   */
+  const isValidHexCode = (hexCode: string) => {
+    const hexRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+    return hexRegex.test(hexCode);
+  };
+
+  const handleClick = () => {
     setShowPicker(!showPicker);
-  }, []);
+  };
 
-  const handleColorChange = useCallback(
-    (color) => {
-      onSelectColor(color.hex);
-    },
-    [color],
-  );
+  const handleColorChange = (color) => {
+    setNewColor(color.hex);
+    onSelectColor(color.hex);
+  };
 
-  const handleHexCodeChange = useCallback(
-    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
+  const handleHexCodeChange = useCallback((event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
+    setNewColor(event.detail.value);
+    if (isValidHexCode(event.detail.value)) {
+      setHexCodeError(''); // Clear any existing error message
       onSelectColor(event.detail.value);
-    },
-    [color],
-  );
+    } else {
+      setHexCodeError(
+        intl.formatMessage({ defaultMessage: 'Invalid hex code', description: 'hex validations messages' }),
+      ); // Set the error message
+    }
+  }, []);
 
-  const handleShowChromePicker = useCallback(() => {
+  const handleShowChromePicker = () => {
     setShowChromePicker(true);
     setShowPicker(false);
-  }, []);
+  };
 
-  const handleClose = useCallback(() => {
+  const handleClose = () => {
     setShowPicker(false);
     setShowChromePicker(false);
-  }, []);
+  };
 
-  const handleCloseChromePicker = useCallback(() => {
+  const handleCloseChromePicker = () => {
```

Comment:
What's the reason for the removal of useCallback to just functions that will get redeclared on rerender?

Reply:
No. It will re-rendered if the function depends on external variable/ computing and passing that variable to another component. Here everything happening in the same component. These are not needed.